nupic/encoders/scalar.py

```diff
@@ -714,3 +716,49 @@ def dump(self):
     print "  nInternal: %d" % self.nInternal
     print "  rangeInternal: %f" % self.rangeInternal
     print "  padding: %d" % self.padding
+
+
+  def __new__(cls, *args, **kwargs):
```

Comment:
Could we instead change `read` to be a classmethod that does this? Or rename this function to something other than `__new__`? It seems odd to pass a capnp proto as the `cls` argument of `__new__`.